woocommerce/woocommerce-ios
# Products - empty view pull to refresh

Closes: #6356


### Description
This PR adds pull to refresh to the empty view controller in `ProductsViewController`.
- Add a pull to refresh callback to empty view when there are no products available.
- Stop adding `EmptyStateViewController` more than once. i.e. Abort if empty view is already presented. (This fix is already present in `OrderListViewController`, `ReviewsViewController`.)

### Testing instructions


- Login to the app with a valid account
- Turn off the internet connection
- Switch to the Products tab. Without internet connection, no data can be fetched, so empty views will be presented shortly afterward.
- Turn on the internet connection.
- Pull down to refresh and observe that products are loaded now.

### Screenshots

https://user-images.githubusercontent.com/524475/171548790-35cba363-c405-4191-bb5d-18d4d5598c7c.MP4



---
